src/wp-includes/ms-blogs.php

```diff
@@ -532,8 +532,10 @@ function switch_to_blog( $new_blog_id, $deprecated = null ) {
 	}
 
 	$wpdb->set_blog_id( $new_blog_id );
-	$GLOBALS['table_prefix'] = $wpdb->get_blog_prefix();
-	$GLOBALS['blog_id']      = $new_blog_id;
+	$GLOBALS['table_prefix']       = $wpdb->get_blog_prefix();
+	$GLOBALS['blog_id']            = $new_blog_id;
+	$GLOBALS['wp_template_path']   = null;
+	$GLOBALS['wp_stylesheet_path'] = null;
```

Comment:
These should be added to the documented globals in the docblock too.

Reply:
Good catch, I added this in 68d1910